imap/imapd.h

```diff
@@ -431,4 +431,9 @@ enum {
 
 extern struct protstream *imapd_out, *imapd_in;
 
+struct client_behavior_registry {
+    int did_compress : 1;   /* client started COMPRESS */
+    int did_notify   : 1;   /* client used NOTIFY */
```

Comment:
Someone said elsewhere recently that we should be using `unsigned` rather than `int` for single-bit struct fields, I don't remember why exactly but it was something about sign extension ambiguity.  I also notice you're logging these fields with `%u` so maybe you meant to do that and forgot to go back.

That said, there's only ever one of these per process, so I don't think the extra instructions to bit shift these values into place are worth the space savings.  I'd just make them both full-size ints, signedness irrelevant.  We're C programmers; we're used to using ints for bools. :)

Reply:
I added the `: 1` at rsto's suggestion, and following some other code we have, so I'm going to stick with that.  I did add the unsigned in an amendment.  I just hadn't thought about it.

Reply:
This program prints -1.  Clang prints also 
```
a.c:9:7: warning: implicit truncation from 'int' to a one-bit wide bit-field changes value from 1 to -1 [-Wsingle-bit-bitfield-constant-conversion]
  b.a = 1;
      ^ ~
1 warning generated.
```
```c
#include <stdio.h>

struct f {
  int a: 1;
};

int main(void) {
  struct f b;
  b.a = 1;
  printf("%i\n", b.a);
  return 0;
}
```